extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -294,7 +295,11 @@ export class SubscriptionService {
             this.fetchEntitlements_(subscriptionPlatform);
           }
       );
-      this.startAuthorizationFlow_();
+
+      // Delegates the platform selection and activation call.
+      const doPlatformSelection = !isStoryDocument(this.ampdoc_.win.document);
+
```

Comment:
Not sure how likely, but in theory it's possible that this code is executed _before_ any of the body children are parsed. Basically, the way this works - the doc services are unblocked as soon as the opening `<body>` tag is parsed.

In other words, if I wrote a 100% correct matcher here, it'd be:

```
isStoryDocument(ampdoc):Promise<boolean> {
  return waitForFirstChildOrFullParse().then(() => {
    return ampdoc.getBody().firstElement &&
       ampdoc.getBody().firstElement.tagName == 'AMP-STORY';
  });
}
```

With this in mind, I'm wondering if it'd be simpler to just check that `<script src="...amp-story-0.1.js">` is declared in the head? This could be done with a sync check.

Reply:
Can we instead check that `document.body.children[0].matches('amp-story[standalone]')`?

Reply:
It'll be `ampdoc.getBody().firstElementChild` and we have to wait for the first element child to actually arrive over the network and be parsed.